script/csv-path-builder.js

```diff
@@ -0,0 +1,161 @@
+const fs = require('fs');
+const path = require('path');
+const cheerio = require('cheerio');
+const Json2csvParser = require('json2csv').Parser;
+
+const rootPath = 'build/localhost';
+
+function getFilesFromDir(dir, fileTypes) {
+  const filesToReturn = [];
+
+  function walkDir(currentPath) {
+    const files = fs.readdirSync(currentPath);
+
+    Object.keys(files).forEach(key => {
+      const curFile = path.join(currentPath, files[key]);
+      if (
+        fs.statSync(curFile).isFile() &&
+        fileTypes.indexOf(path.extname(curFile)) !== -1
+      ) {
+        filesToReturn.push(curFile.replace(dir, ''));
+      } else if (fs.statSync(curFile).isDirectory()) {
+        walkDir(curFile);
+      }
+    });
+  }
+
+  walkDir(dir);
+
+  return filesToReturn;
+}
+
+const fileList = getFilesFromDir(rootPath, ['.html']);
+
+const getPageQuery = filePath => {
+  const fileContent = fs.readFileSync(filePath);
+  const data = fileContent.toString();
+
+  return cheerio.load(data);
+};
+
+const getTargetTitle = href => {
+  if (href) {
+    if (href.startsWith('/')) {
+      let filePath = `${rootPath}${href}/index.html`;
+
+      if (href === '/') {
+        filePath = `${rootPath}/index.html`;
+      }
+
+      if (href.includes('.pdf')) {
+        if (fs.existsSync(`${rootPath}${href}`)) {
+          return `This is a ${path.extname(href)} file`;
+        }
+
+        return 'Page does not exist';
+      }
+
+      if (href.includes('/introduction')) {
+        filePath = `${rootPath}${href.replace('/introduction', '')}/index.html`;
+        const $ = getPageQuery(filePath);
+
+        return $('title')
+          .text()
+          .trim();
+      }
+
+      if (fs.existsSync(filePath)) {
+        const $ = getPageQuery(filePath);
+
+        return $('title')
+          .text()
+          .trim();
+      }
+
+      if (filePath.includes('#')) {
+        filePath = `${rootPath}${href.split('/#')[0]}/index.html`;
+
+        if (fs.existsSync(filePath)) {
+          const $ = getPageQuery(filePath);
+
+          return $('title')
+            .text()
+            .trim();
+        }
+      }
+
+      return `Page does not exist`;
+    }
+
+    if (href.includes('localhost')) {
+      if (href.includes('.asp')) {
+        return 'External Link';
```

Comment:
ASP pages in our file list are redirects to internal pages.

Reply:
These pages don't exist in the build directory, so I will just label these `Original VA.gov page (Proxy)`.